rust/src/encodings/plain.rs

```diff
@@ -65,70 +65,99 @@ impl<'a> PlainEncoder<'a> {
 
     /// Encode an array of a batch.
     /// Returns the offset of the metadata
-    pub async fn encode(&mut self, array: &dyn Array) -> Result<usize> {
+    pub async fn encode(&mut self, array: &[&dyn Array]) -> Result<usize> {
         self.encode_internal(array, self.data_type).await
     }
 
     #[async_recursion]
-    async fn encode_internal(&mut self, array: &dyn Array, data_type: &DataType) -> Result<usize> {
+    async fn encode_internal(
+        &mut self,
+        array: &[&'async_recursion dyn Array],
```

Comment:
Could you verify that this lifetime have the same name as `[async_recurisve]`?

Usually a lifetime is single character like `'a`, `'b`, and etc.